### PR TITLE
AndroidTV Config Flow

### PR DESCRIPTION
**Pull request by ollo69**
## Proposed change

Remove yaml configuration documentation
Add config flow documentation

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/54444
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/androidtv.markdown [hunk 1]
```diff
@@ -34,140 +36,19 @@ For Fire TV devices, the instructions are as follows:
...
-  type: string
-screencap:
-  description: Determines if album art should be pulled from what is shown onscreen.
-  required: false
-  default: true
-  type: boolean
-{% endconfiguration %}
-
-### Full Configuration
+It is possible to change some behaviors through the integration options. These can be changed at **AndroidTV** -> **Configure** on the Integrations page.
```
Comment (frenck) [inline source/_integrations/androidtv.markdown]: We have helpers for these as well, see the `cast` documentation for an example

File: source/_integrations/androidtv.markdown [hunk 2]
```diff
@@ -179,14 +60,22 @@ When connecting to your device for the first time, a dialog will appear on your
 
 ### 1. Python ADB Implementation
 
-The default approach is to connect to your device using the `adb-shell` Python package. As of Home Assistant 0.101, if a key is needed for authentication and it is not provided by the `adbkey` configuration option, then Home Assistant will generate a key for you.
+The default approach is to connect to your device using the `adb-shell` Python package. As of Home Assistant 0.101, if a key is needed for authentication and it is not provided by the `ADB Key` setup option, then Home Assistant will generate a key for you.
+
+<div class='note'>
+To be able to provide `ADB Key` on integration setup, you need to enable [advanced mode](https://www.home-assistant.io/blog/2019/07/17/release-96/#advanced-mode).
+</div>
```
Comment (frenck) [inline source/_integrations/androidtv.markdown]: Adjustment to the link and an empty line before after language switching to help the Markdown parse to switch HTML/Markdown context.

```suggestion
<div class='note'>

To be able to provide `ADB Key` on integration setup, you need to enable [advanced mode](/blog/2019/07/17/release-96/#advanced-mode).

</div>
```

File: source/_integrations/androidtv.markdown [hunk 3]
```diff
@@ -179,14 +60,22 @@ When connecting to your device for the first time, a dialog will appear on your
...
 
 Prior to Home Assistant 0.101, this approach did not work well for newer devices. Efforts have been made to resolve these issues, but if you experience problems then you should use the ADB server option.
 
 ### 2. ADB Server
 
 The second option is to use an ADB server to connect to your Android TV and Fire TV devices.
 
+<div class='note'>
+To configure ADB server on integration setup, you need to enable [advanced mode](https://www.home-assistant.io/blog/2019/07/17/release-96/#advanced-mode).
+</div>
```
Comment (frenck) [inline source/_integrations/androidtv.markdown]:
```suggestion
<div class='note'>

To configure ADB server on integration setup, you need to enable [advanced mode](/blog/2019/07/17/release-96/#advanced-mode).

</div>
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/androidtv.markdown [hunk 4]
```diff
@@ -34,140 +36,25 @@ For Fire TV devices, the instructions are as follows:
...
-    adb_server_ip: 127.0.0.1
-    adb_server_port: 5037
-    get_sources: false
-```
+{% include integrations/config_flow.md %}
+
+{% include integrations/option_flow.md %}
+{% configuration_basic %}
+Configure Applications List:
+  description: Here you can define a list of applications where the keys are app IDs and the values are app names that will be displayed in the UI. If a name is not provided, the app will never be shown in the sources list. ([These app names](https://github.com/JeffLIrion/python-androidtv/blob/748d6b71cad611c624ef7526d9928431167531a3/androidtv/constants.py#L290-L308) are configured in the backend package and do not need to be included in your configuration).
```
Comment (JeffLIrion) [inline source/_integrations/androidtv.markdown]: > If a name is not provided, the app will never be shown in the sources list.

That is only true if `exclude_unnamed_apps` (or its successor) is true. And I believe it is false by default.

File: source/_integrations/androidtv.markdown [hunk 5]
```diff
@@ -179,14 +66,26 @@ When connecting to your device for the first time, a dialog will appear on your
 
 ### 1. Python ADB Implementation
 
-The default approach is to connect to your device using the `adb-shell` Python package. As of Home Assistant 0.101, if a key is needed for authentication and it is not provided by the `adbkey` configuration option, then Home Assistant will generate a key for you.
+The default approach is to connect to your device using the `adb-shell` Python package. As of Home Assistant 0.101, if a key is needed for authentication and it is not provided by the `ADB Key` setup option, then Home Assistant will generate a key for you.
```
Comment (JeffLIrion) [inline source/_integrations/androidtv.markdown:71]: Is providing an ADB key a hidden/advanced configuration option? After all, it's easier and simpler to just let HA generate a key.

Comment (ollo69) [reply source/_integrations/androidtv.markdown:71]: Yes is advanced option, and is specified in the next chapter `Python ADB Implementation`

File: source/_integrations/androidtv.markdown [hunk 6]
```diff
@@ -34,159 +36,58 @@ For Fire TV devices, the instructions are as follows:
...
-    adb_server_ip: 127.0.0.1
-    adb_server_port: 5037
-    get_sources: false
-```
+{% include integrations/config_flow.md %}
+
+{% include integrations/option_flow.md %}
+{% configuration_basic %}
+Configure Applications List:
+  description: Here you can define a list of applications where the keys are app IDs and the values are app names that will be displayed in the UI. If a name is not provided and the option `Exclude apps with unknown name` is enabled, the app will never be shown in the sources list. ([These app names](https://github.com/JeffLIrion/python-androidtv/blob/748d6b71cad611c624ef7526d9928431167531a3/androidtv/constants.py#L290-L308) are configured in the backend package and do not need to be included in your configuration).
```
Comment (JeffLIrion) [inline source/_integrations/androidtv.markdown]:
```suggestion
  description: Here you can define a list of applications where the keys are app IDs and the values are app names that will be displayed in the UI. If a name is not provided and the option `Exclude apps with unknown name` is enabled, the app will never be shown in the sources list. ([These app names](https://github.com/JeffLIrion/python-androidtv/blob/v0.0.60/androidtv/constants.py#L396-L482) are configured in the backend package and do not need to be included in your configuration).
```

Comment (frenck) [reply source/_integrations/androidtv.markdown]: Honestly, we should not be linking to source code from end-user documentation AT ALL.

Comment (ollo69) [reply source/_integrations/androidtv.markdown]: There are other points where should code is referenced. should we remove everywhere? What you suggest, to generally link to the library repository?

Comment (frenck) [reply source/_integrations/androidtv.markdown]: > What you suggest, to generally link to the library repository?

I suggest that the needed configuration options are documented in the Home Assistant documentation.

File: source/_integrations/androidtv.markdown [hunk 7]
```diff
@@ -34,159 +36,58 @@ For Fire TV devices, the instructions are as follows:
...
 </div>
 
 ### 1. Python ADB Implementation
 
-The default approach is to connect to your device using the `adb-shell` Python package. As of Home Assistant 0.101, if a key is needed for authentication and it is not provided by the `adbkey` configuration option, then Home Assistant will generate a key for you.
+The default approach is to connect to your device using the `adb-shell` Python package. As of Home Assistant 0.101, if a key is needed for authentication and it is not provided by the `ADB Key` setup option, then Home Assistant will generate a key for you.
+
+<div class='note'>
+
+To be able to provide `ADB Key` on integration setup, you need to enable [advanced mode](https://www.home-assistant.io/blog/2019/07/17/release-96/#advanced-mode).
```
Comment (frenck) [inline source/_integrations/androidtv.markdown]:
```suggestion
To be able to provide `ADB Key` on integration setup, you need to enable [advanced mode](/blog/2019/07/17/release-96/#advanced-mode).
```

File: source/_integrations/androidtv.markdown [hunk 8]
```diff
@@ -34,159 +36,58 @@ For Fire TV devices, the instructions are as follows:
...
 
 Prior to Home Assistant 0.101, this approach did not work well for newer devices. Efforts have been made to resolve these issues, but if you experience problems then you should use the ADB server option.
 
 ### 2. ADB Server
 
 The second option is to use an ADB server to connect to your Android TV and Fire TV devices.
 
+<div class='note'>
+
+To configure ADB server on integration setup, you need to enable [advanced mode](https://www.home-assistant.io/blog/2019/07/17/release-96/#advanced-mode).
```
Comment (frenck) [inline source/_integrations/androidtv.markdown]:
```suggestion
To configure ADB server on integration setup, you need to enable [advanced mode](/blog/2019/07/17/release-96/#advanced-mode).
```

**Review by frenck — APPROVED** [review]
Thanks, @ollo69 👍

**Pull request merged**